enhancements/machine-api/dedicated-instances.md

```diff
@@ -0,0 +1,196 @@
+```
+A host group is a resource that represents a collection of dedicated hosts. You create a host group in a region and an availability zone, and add hosts to it.
+
+A host is a resource, mapped to a physical server in an Azure data center. The physical server is allocated when the host is created. A host is created within a host group. A host has a SKU describing which VM sizes can be created. Each host can host multiple VMs, of different sizes, as long as they are from the same size series.
+
+When creating a VM in Azure, you can select which dedicated host to use for your VM. You have full control as to which VMs are placed on your hosts.
+```
+
+The problem here are standard quotas: for host of type `DSv3-Type1` we can create only 32 VM of type `Standard_D2s_v3`(default type for worker VMs). To request a quota increase, the users are required to create a support request. This part should be well documented
+
+The required API change is adding host name field `Host` to provider spec.
+
+```
+type AzureMachineProviderConfig struct {
+  // existing fields
+  ...
+
+  // Host name of physical server that hosts the virtual machine
+  // +optional
+  Host string
+}
+```
+
+#### GCP
+
+GCP requires `Node Templates` and `Node Groups` to be able to create dedicated instances. [GCP documentation](https://cloud.google.com/compute/docs/nodes/sole-tenant-nodes).
+
+```
+Node templates
+
+A node template is a regional resource that defines the properties of each node in a node group. 
+
+Node groups and VM provisioning
+
+Sole-tenant node templates define the properties of a node group, and you must create a node template before creating a node group in a Google Cloud zone. When you create a group, specify the maintenance policy for VM instances on the node group, and the number of nodes for the node group. A node group can have zero or more nodes; for example, you can reduce the number of nodes in a node group to zero when you don't need to run any VM instances on nodes in the group, or you can enable the node group autoscaler to manage the size of the node group automatically.
+```
+
+In order to be able to create a VM on a dedicated host we should introduce `NodeGroup` API field to provider spec.
+
+We should document that node groups have resource capacities which limit the number of VMs, unless the node group autoscaler is enabled.
+
+```
+type GCPMachineProviderConfig struct {
+  // existing fields
+  ...
+
+  // NodeGroup name of node group that hosts the virtual machine
+  // +optional
+  NodeGroup string
+}
+```
+
+### Risks and Mitigations
+
+- Different quotas issues in Azure. See [this](#azure) section for more details.
+- Azure and GCP will require some [configurations](#infrastructure-needed) on cloud provider side to be made before creating a dedicated instance.
+
+#### Autoscaling
+
+Autoscaling dedicated instances can be a problem because dedicated hosts have quotas and limits on provider side. We should provide good documentation here.
```

Comment:
is there any more detail we could add here, what does the user need to be aware of?

Reply:
nothing more that I know about